### PR TITLE
[Issue #5891] Implement MessageStore interface for tiered store

### LINKED ISSUE #5891: [RIP-57] Implement MessageStore interface for tiered store
link to #5754 

This issue aims to do these jobs. 

- Implement MessageStore interface
- Implement metrics
- Add unit tests

### PR DESCRIPTION
**Pull request by ShadowySpirits**
close #5891

### CODE REVIEW

File: tieredstore/src/main/java/org/apache/rocketmq/store/tiered/TieredDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+import org.apache.rocketmq.logging.org.slf4j.LoggerFactory;
+import org.apache.rocketmq.store.CommitLogDispatcher;
+import org.apache.rocketmq.store.ConsumeQueue;
+import org.apache.rocketmq.store.DispatchRequest;
+import org.apache.rocketmq.store.MessageStore;
+import org.apache.rocketmq.store.SelectMappedBufferResult;
+import org.apache.rocketmq.store.tiered.common.AppendResult;
+import org.apache.rocketmq.store.tiered.common.TieredMessageStoreConfig;
+import org.apache.rocketmq.store.tiered.common.TieredStoreExecutor;
+import org.apache.rocketmq.store.tiered.container.TieredContainerManager;
```
Comment (aaron-ai) [inline tieredstore/src/main/java/org/apache/rocketmq/store/tiered/TieredDispatcher.java:45]: Could we rename `rocketmq.store` to `rocketmq.tieredstore`?

File: tieredstore/src/main/java/org/apache/rocketmq/store/tiered/container/TieredMessageQueueContainer.java [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+        }
+
+        return consumeQueue.append(request.getCommitLogOffset(), request.getMsgSize(), request.getTagsCode(), request.getStoreTimestamp(), commit);
+    }
+
+    public AppendResult appendIndexFile(DispatchRequest request) {
+        if (closed) {
+            return AppendResult.FILE_CLOSED;
+        }
+//        AppendResult result = indexFile.append(messageQueue, request.getOffsetId(), request.getCommitLogOffset(), request.getMsgSize(), request.getStoreTimestamp());
```
Comment (aaron-ai) [inline tieredstore/src/main/java/org/apache/rocketmq/store/tiered/container/TieredMessageQueueContainer.java:374]: Add more comments here.

**Comment by aaron-ai** [comment]
About the part of calling APIs of the specific storage system, there should be an independent interface class including the method of  `TieredFileSegment#commit0` and `TieredFileSegment#read0`. The single responsibility principle is essential.

File: tieredstore/src/main/java/org/apache/rocketmq/store/tiered/TieredDispatcher.java (hunk 1 above)
Comment (lollipopjin) [reply tieredstore/src/main/java/org/apache/rocketmq/store/tiered/TieredDispatcher.java:45]: Here means 'rocketmq.store.tiered' changes to 'rocketmq.tieredstore'

File: tieredstore/src/main/java/org/apache/rocketmq/store/tiered/metrics/TieredStoreMetricsManager.java [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            .setName(HISTOGRAM_UPLOAD_BYTES)
+            .build();
+
+        InstrumentSelector downloadBufferSizeSelector = InstrumentSelector.builder()
+            .setType(InstrumentType.HISTOGRAM)
+            .setName(HISTOGRAM_DOWNLOAD_BYTES)
+            .build();
+
+        View bufferSizeView = View.builder()
+            .setAggregation(Aggregation.explicitBucketHistogram(Arrays.asList(1024d, 10d * 1024, 100d * 1024, 1024d * 1024, 10d * 1024 * 1024, 32d * 1024 * 1024, 50d * 1024 * 1024, 100d * 1024 * 1024)))
```
Comment (lollipopjin) [inline tieredstore/src/main/java/org/apache/rocketmq/store/tiered/metrics/TieredStoreMetricsManager.java:131]: Here can we use     
private static final double KB = 1024d;
1 * KB, 10 * KB ....  
to improve the readability?

**Comment by ShadowySpirits** [comment]
> About the part of calling APIs of the specific storage system, there should be an independent interface class including the method of `TieredFileSegment#commit0` and `TieredFileSegment#read0`. The single responsibility principle is essential.

I have moved `TieredFileSegment` to tieredstore.provider package and make it implement `TieredStoreBackendProvider` interface.

**Review by lollipopjin — APPROVED** [review]
LGTM now.

**Pull request merged**